Dont confirm switch worktree option

KarstenRa opened the pull request:
Fixes #6864


## Proposed changes

- Option in the confirmations setting menu to not ask for a confirmation when switching the worktree


## Screenshots
![switchWorktree](https://user-images.githubusercontent.com/26045339/66057783-96f44c00-e539-11e9-8ad7-6e4bfcaf95f0.PNG)



## Test methodology <!-- How did you ensure quality? -->

Manual


## Test environment(s)
- Git Extensions 3.3.0
- Build 429a4f06e30478c41bdfaf1f501de175b82b8126 (Dirty)
- Git 2.22.0.windows.1 (recommended: 2.23.0 or later)
- Microsoft Windows NT 10.0.17134.0
- .NET Framework 4.7.3460.0
- DPI 120dpi (125% scaling)


----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

pmiossec commented:
You have to translate some strings. See https://github.com/gitextensions/gitextensions/wiki/Translations

KarstenRa commented:
> 
> 
> You have to translate some strings. See https://github.com/gitextensions/gitextensions/wiki/Translations

I keep getting "File not found errors" on executing
` .\Build\transifex\UpdateLocalEnglishTranslations.ps1  -Configuration Debug`

```
Updating the English translation...
Start-Process : File not found
In ...\Build\transifex\UpdateLocalEnglishTranslations.ps1:18
+         Start-Process -FilePath "$src\TranslationApp.exe" -ArgumentLi ...
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [Start-Process], InvalidOperationException
    + FullyQualifiedErrorId : InvalidOperationException,Microsoft.PowerShell.Commands.StartProcessCommand
```

Any suggestions?

pmiossec commented:
Either you are not in the good directory or the TranslationApp project is
not built. Try a rebuild all (or find out why...)

>

KarstenRa commented:
> 
> Either you are not in the good directory or the TranslationApp project is not built. Try a rebuild all (or find out why...)
> […](#)

Thanks, rebuild did it

pmiossec commented on GitUI/CommandsDialogs/WorktreeDialog/FormManageWorktree.cs:
```diff
@@ -207,8 +207,8 @@ private void Worktrees_CellClick(object sender, DataGridViewCellEventArgs e)
-                if (MessageBox.Show(this, "Are you sure you want to switch to this worktree?", "Open a worktree",
-                    MessageBoxButtons.YesNo) == DialogResult.Yes)
+                if (AppSettings.DontConfirmSwitchWorktree || MessageBox.Show(this,
+                        "Are you sure you want to switch to this worktree?", "Open a worktree", MessageBoxButtons.YesNo) == DialogResult.Yes)
```
Could you have the kindness to translate these strings? I know it is not your fault but it is the good moment to do it...

KarstenRa replied:
Sure, I gave it a try ;)

pmiossec commented:
That's just define and use a 'TranslatedString' (see usage elsewhere)

On Thu, Oct 3, 2019, 17:20 Karsten <notifications@github.com> wrote:

> *@KarstenRa* commented on this pull request.
> ------------------------------
>
> In GitUI/CommandsDialogs/WorktreeDialog/FormManageWorktree.cs
> <https://github.com/gitextensions/gitextensions/pull/7230#discussion_r331100521>
> :
>
> > @@ -207,8 +207,8 @@ private void Worktrees_CellClick(object sender, DataGridViewCellEventArgs e)
>
>              if (e.ColumnIndex == 5)
>              {
> -                if (MessageBox.Show(this, "Are you sure you want to switch to this worktree?", "Open a worktree",
> -                    MessageBoxButtons.YesNo) == DialogResult.Yes)
> +                if (AppSettings.DontConfirmSwitchWorktree || MessageBox.Show(this,
> +                        "Are you sure you want to switch to this worktree?", "Open a worktree", MessageBoxButtons.YesNo) == DialogResult.Yes)
>
> Sure, I gave it a try ;)
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/gitextensions/gitextensions/pull/7230?email_source=notifications&email_token=AADQLJADCCDFZK4PMOEUEW3QMYESFA5CNFSM4I4X4UHKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCGZR7KQ#discussion_r331100521>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AADQLJG77RQOPD5RDU74O6LQMYESFANCNFSM4I4X4UHA>
> .
>

KarstenRa commented:
> 
> That's just define and use a 'TranslatedString' (see usage elsewhere)
> […](#)
> On Thu, Oct 3, 2019, 17:20 Karsten ***@***.***> wrote: *@KarstenRa* commented on this pull request. ------------------------------ In GitUI/CommandsDialogs/WorktreeDialog/FormManageWorktree.cs <[#7230 (comment)](https://github.com/gitextensions/gitextensions/pull/7230#discussion_r331100521)> : > @@ -207,8 +207,8 @@ private void Worktrees_CellClick(object sender, DataGridViewCellEventArgs e) if (e.ColumnIndex == 5) { - if (MessageBox.Show(this, "Are you sure you want to switch to this worktree?", "Open a worktree", - MessageBoxButtons.YesNo) == DialogResult.Yes) + if (AppSettings.DontConfirmSwitchWorktree || MessageBox.Show(this, + "Are you sure you want to switch to this worktree?", "Open a worktree", MessageBoxButtons.YesNo) == DialogResult.Yes) Sure, I gave it a try ;) — You are receiving this because you commented. Reply to this email directly, view it on GitHub <#7230?email_source=notifications&email_token=AADQLJADCCDFZK4PMOEUEW3QMYESFA5CNFSM4I4X4UHKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCGZR7KQ#discussion_r331100521>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AADQLJG77RQOPD5RDU74O6LQMYESFANCNFSM4I4X4UHA> .

Not sure what you mean, I updated the PR and used a TranslationString class variable. After an update the English.xlf file was updated with the new (Switch Worktree) string and I added it to the PR. Don't know what I am missing here...

pmiossec commented:
I mean doing the exact same thing for the 2 others hardcoded strings.

>

KarstenRa commented:
> 
> I mean doing the exact same thing for the 2 others hardcoded strings.
> […](#)

Still not sure if you did not see that I updated my commit or if I am confused... Please point out class and line of the two hardcoded strings you are reffering to.

pmiossec commented:
Sorry. Indeed, I did not saw it.
Perfect!
